### PR TITLE
chore(module:*): enable `strictTemplates`

### PR DESCRIPTION
**Pull request by hsuanxyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/checkbox/checkbox-group.component.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export interface NzCheckBoxOptionInterface {
       class="ant-checkbox-group-item"
       *ngFor="let o of options; trackBy: trackByOption"
       [nzDisabled]="o.disabled || nzDisabled"
-      [(nzChecked)]="o.checked"
+      [(nzChecked)]="!!o.checked"
```
Comment (vthinkxie) [inline components/checkbox/checkbox-group.component.ts:33]: would cause bug here

Comment (hsuanxyz) [reply components/checkbox/checkbox-group.component.ts:33]: https://github.com/angular/angular/issues/36551

File: components/notification/notification.component.ts [hunk 2]
```diff
@@ -21,8 +21,8 @@ import { NzNotificationData } from './typings';
   template: `
     <div
       class="ant-notification-notice ant-notification-notice-closable"
-      [ngStyle]="instance.options?.nzStyle"
-      [ngClass]="instance.options?.nzClass"
+      [ngStyle]="instance.options ? instance.options.nzStyle! : null"
```
Comment (wendellhu95) [inline components/notification/notification.component.ts:24]: Could it be simplified as `[ngStyle]="instance?.options || null`?

File: components/time-picker/demo/basic.ts [hunk 3]
```diff
@@ -3,7 +3,10 @@ import { Component } from '@angular/core';
 @Component({
   selector: 'nz-demo-time-picker-basic',
   template: `
-    <nz-time-picker [(ngModel)]="time" [nzDefaultOpenValue]="defaultOpenValue"></nz-time-picker>
+    <nz-time-picker [(ngModel)]="time" [nzDefaultOpenValue]="defaultOpenValue" [nzSuffixIcon]="t"></nz-time-picker>
+    <ng-template #t>
+      X
+    </ng-template>
```
Comment (wenqi73) [inline components/time-picker/demo/basic.ts:9]: 测试代码，需要改回去

**Comment by wenqi73** [comment]
LGTM (partially).

**Pull request merged**